src/main/java/com/github/cameltooling/lsp/internal/completion/CamelComponentSchemesCompletionsFuture.java

```diff
@@ -41,6 +44,7 @@ public CamelComponentSchemesCompletionsFuture(String filterText) {
 					CompletionItem completionItem = new CompletionItem(componentModel.getSyntax());
 					completionItem.setDocumentation(componentModel.getDescription());
 					completionItem.setDeprecated(Boolean.valueOf(componentModel.getDeprecated()));
+					completionItem.setData(CompletionResolverUtils.getCompletionResolverDataForUriInstance(uriElement, completionItem.getLabel()));
```

Comment:
why did you choose to pass Data to the CompletionItem and provide a textEdit only on resolveCompletionItem?
it seems to me that in com.github.cameltooling.lsp.internal.completion.CompletionResolveHandler.resolve(CompletionItem) it is just retrieving values from the data to create a textEdit without doing any computations. I don't see the interest then.

Reply:
I was basically just imitating what JDT.LS was doing. If you think there is a better way then we can improve it later on.

Reply:
the difference is that in JDt they are doing a lot more things that is taking time to compute in the resolve method https://github.com/eclipse/eclipse.jdt.ls/blob/c739690a58d27a9077a0b7d2daece258ba64cb11/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/CompletionResolveHandler.java#L70
i think it is is useless for us for now.
Not doing it is greatly simplifying code, I really think it is worthy to do it in this PR